### PR TITLE
Create domain name resolver plugin

### PR DESCRIPTION
**Pull request by nemski**
Takes a nameserver and resolve_address as input and raises an alert if resolution fails

### CODE REVIEW

File: name_resolution/name_resolver.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      default: google.com
+  EOS
+
+  def build_report
+    resolver = Resolv::DNS.new(:nameserver => [option(:nameserver)])
+
+    begin
+      result = resolver.getaddress(option(:resolve_address))
+    rescue Resolv::ResolvError => err
+      alert('Failed to resolve', err)
```
Comment (ejoubaud) [inline name_resolution/name_resolver.rb]: Will this systematically trigger an alert if the namespace can't resolve once? Is this something we expect to happen rarely enough that we're happy getting an alert each time? Would we have more flexibility with a 0-1 metric to graph/set triggers over time?

File: name_resolution/name_resolver.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+require 'resolv'
+
+class NameResolver < Scout::Plugin
+  OPTIONS=<<-EOS
+    nameserver:
+      default: 8.8.8.8
```
Comment (denmat) [inline name_resolution/name_resolver.rb]: This isn't the nameserver we want to test against -- localhost (127.0.0.1) would be better.

Comment (nemski) [inline name_resolution/name_resolver.rb]: Yep just putting in some sane defaults

**Comment by denmat** [comment]
I agree with @ejoubaud -- metrics over time much better.  Maybe look at the ntp check for how that is done https://github.com/scoutapp/scout-plugins/blob/master/ntp_time/ntp_time.rb

**Comment by nemski** [comment]
Ok changed to use `report` instead of `alert`, but I hit another snag. When DNS resolution times out it raises a PluginTimeoutError, as the default plugin timeout is 60 seconds and DNS timeouts take longer. Only Ruby >= 2.0 supports setting a custom timeout on Resolv::DNS. I've raised an issue with Scout on how we set custom plugin timeouts (documentation indicates its supported but can't get it to work). Shall we ship this in the mean time? I've no idea what the graphs will look if there is a timeout like but at least we'll have something.

**Comment by denmat** [comment]
I think a 60s timeout is too long on dns anyway -- I would expect it to resolve much quicker than that. 

Let's ship and verify how the metrics look.

**Comment by nemski** [comment]
Actually wrapping it in Timeout.timeout works! (have to initialize the resolver and run the query within the Timeout block)

**Comment by ejoubaud** [comment]
Late +1, nice one!

**Pull request merged**